optimism/Interpolants.py

```diff
@@ -41,6 +48,12 @@
             of spatial dimensions. Line elements are an exception, which
             have shape ``(nPts, nNdodes)``.
     """
+    values: Float[Array, "nq nn"]
+    gradients: Float[Array, "nq nn nd"]
+
+    def __iter__(self):
+        yield self.values
+        yield self.gradients
```

Comment:
Is an iterator needed for this class?

Reply:
@ralberd this is not necessary. I added it because there are several places 8n the code that unpacks the interpolants as a tuple. This is just a trick to get that syntactic sugar.